instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/SetRouteOrderTest.kt

```diff
@@ -122,7 +124,7 @@ class SetRouteOrderTest : BaseTest<BasicNavigationViewActivity>(
                     .trafficExpressionProvider!!
                     .generateExpression()
                     .contents as ArrayList<*>
-                assertEquals(shortRoute, primaryRoute)
+                assertTrue(shortRoute.isSameRoute(primaryRoute))
```

Comment:
Since we control the route references that are passed to the API, we are setting the right expectation that they should be equal here. Relaxing the check might just mask the issue. Do we know what mutates the internals of the route? Or maybe it's the fact that the long route is returned here while short should be?

Reply:
You're right that long route was being returned. I was grasping at straws here.  The flakiness surfaced recently. I think it's related to the caching that was recently introduced with the `NativeRouteParserWrapper` but even trying to adjust for that in the test hasn't been entirely successful.  The test always passes for me locally which makes solving this all the more difficult.